x/uniswap/internal/keeper/keeper.go

```diff
@@ -0,0 +1,146 @@
+	return ctx.Logger().With("module", types.ModuleName)
+}
+
+// CreateReservePool initializes a new reserve pool for the new denomination
+func (keeper Keeper) CreateReservePool(ctx sdk.Context, denom string) {
+	store := ctx.KVStore(keeper.storeKey)
+	key := GetReservePoolKey(denom)
+	bz := store.Get(key)
+	if bz != nil {
+		panic(fmt.Sprintf("reserve pool for %s already exists", denom))
+	}
+
+	store.Set(key, keeper.encode(sdk.ZeroInt()))
+}
+
+// GetUNIForAddress returns the total UNI at the provided address
+func (keeper Keeper) GetUNIForAddress(ctx sdk.Context, addr sdk.AccAddress) sdk.Int {
+	var balance sdk.Int
+
+	store := ctx.KVStore(keeper.storeKey)
+	key := GetUNIBalancesKey(addr)
+	bz := store.Get(key)
+	if bz != nil {
+		balance = keeper.decode(bz)
+	}
+
+	return balance
+}
+
+// SetUNIForAddress sets the provided UNI at the given address
+func (keeper Keeper) SetUNIForAddress(ctx sdk.Context, amt sdk.Int, addr sdk.AccAddress) {
+	store := ctx.KVStore(keeper.storeKey)
+	key := GetUNIBalancesKey(addr)
+	store.Set(key, keeper.encode(amt))
+}
+
+// GetTotalUNI returns the total UNI currently in existence
+func (keeper Keeper) GetTotalUNI(ctx sdk.Context) sdk.Int {
+	store := ctx.KVStore(keeper.storeKey)
+	bz := store.Get(TotalUNIKey)
+	if bz == nil {
+		return sdk.ZeroInt()
+	}
+
+	totalUNI := keeper.decode(bz)
+	return totalUNI
+}
+
+// SetTotalLiquidity sets the total UNI
+func (keeper Keeper) SetTotalUNI(ctx sdk.Context, totalUNI sdk.Int) {
+	store := ctx.KVStore(keeper.storeKey)
+	store.Set(TotalUNIKey, keeper.encode(totalUNI))
+}
+
+// GetReservePool returns the total balance of an reserve pool at the provided denomination
+func (keeper Keeper) GetReservePool(ctx sdk.Context, denom string) sdk.Int {
+	store := ctx.KVStore(keeper.storeKey)
+	bz := store.Get(GetReservePoolKey(denom))
+	if bz == nil {
+		panic(fmt.Sprintf("reserve pool for %s does not exist"), denom)
+	}
+	return keeper.decode(bz)
+}
+
+// GetNativeDenom returns the native denomination for this module from the global param store
+func (keeper Keeper) GetNativeDenom(ctx sdk.Context) (nativeDenom string) {
+	keeper.paramSpace.Get(ctx, types.KeyNativeDenom, &nativeDenom)
+	return
+}
+
+// GetFeeParams returns the current FeeParams from the global param store
+func (keeper Keeper) GetFeeParams(ctx sdk.Context) (feeParams sdk.Dec) {
+	keeper.paramSpace.Get(ctx, types.KeyFee, &feeParams)
+	return
+}
+
+func (keeper Keeper) SetFeeParam(ctx sdk.Context, feeParams sdk.Dec) {
+	keeper.paramSpace.Set(ctx, types.KeyFee, &feeParams)
+}
+
+// -----------------------------------------------------------------------------
+// Misc.
+
+func (keeper Keeper) decode(bz []byte) (balance sdk.Int) {
+	err := keeper.cdc.UnmarshalBinaryBare(bz, &balance)
+	if err != nil {
+		panic(err)
+	}
+	return
+}
+
+func (keeper Keeper) encode(balance sdk.Int) (bz []byte) {
+	bz, err := keeper.cdc.MarshalBinaryBare(balance)
+	if err != nil {
+		panic(err)
+	}
+	return
+}
```

Comment:
I guess it's better to use `MustUnmarshalBinaryBare` and `MustMarshalBinaryBare` instead. They take care of the panic part. This way the code gets reduced to one line and we don't even require the two methods.